packages/core/src/Transform.ts

```diff
@@ -252,10 +252,19 @@ export class Transform extends Component {
       this._localMatrix.copyFrom(value);
     }
 
+    // @ts-ignore
+    this._position._onValueChanged = this._rotationQuaternion._onValueChanged = this._scale._onValueChanged = null;
     this._localMatrix.decompose(this._position, this._rotationQuaternion, this._scale);
+    // @ts-ignore
+    this._position._onValueChanged = this._onPositionChanged;
+    // @ts-ignore
+    this._rotationQuaternion._onValueChanged = this._onRotationQuaternionChanged;
+    // @ts-ignore
+    this._scale._onValueChanged = this._onScaleChanged;
 
     this._setDirtyFlagTrue(TransformModifyFlags.LocalEuler);
     this._setDirtyFlagFalse(TransformModifyFlags.LocalMatrix);
+    this._setDirtyFlagFalse(TransformModifyFlags.LocalQuat);
```

Comment:
merge this `this._setDirtyFlagFalse(TransformModifyFlags.LocalMatrix | TransformModifyFlags.LocalQuat);`